lib/src/util/dart_type_utilities.dart

```diff
@@ -188,30 +188,47 @@ class DartTypeUtilities {
   static bool isNullLiteral(Expression expression) =>
       expression?.unParenthesized is NullLiteral;
 
-  static PropertyAccessorElement lookUpGetter(MethodDeclaration node) =>
-      (node.parent as ClassOrMixinDeclaration)
-          .declaredElement
+  static PropertyAccessorElement lookUpGetter(MethodDeclaration node) {
+    final parent = node.parent;
+    if (parent is ClassOrMixinDeclaration) {
```

Comment:
Given that we ask `declaredElement`, it might be better to use `node.declaredElement.enclosingElement` and then check if it is `ClassElement` or `ExtensionElement`.

Reply:
Good idea. 👍